backend/database/versions/9023d0bf61ab_harmonize_schema_state_prod.py

```diff
@@ -0,0 +1,86 @@
+"""harmonize_schema_state_prod
+
+Revision ID: 9023d0bf61ab
+Revises: d9fb29897244
+Create Date: 2020-08-28 15:00:56.262160
+
+"""
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+# revision identifiers, used by Alembic.
+revision = "9023d0bf61ab"
+down_revision = "d9fb29897244"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    # ### commands auto generated by Alembic - please adjust! ###
+    op.drop_table("dataset_deployment")
+    op.alter_column("dataset", "project_id", existing_type=sa.VARCHAR(), nullable=False)
+    op.alter_column("dataset", "project_status", existing_type=sa.VARCHAR(), nullable=False)
+    op.alter_column("dataset", "revision", existing_type=sa.INTEGER(), nullable=True)
+    op.alter_column("deployment_directory", "created_at", existing_type=postgresql.TIMESTAMP(), nullable=False)
```

Comment:
Any `created_at` and `updated_at` fields should have a `server_default=func.now()` I believe so that it gets populated by the database and not us.

(from sqlalchemy import func)

Reply:
Yeah, none of our code should ever populate it, and in fact postgres gives a DEFAULT=now() of some sort for updated_at. But it's still part of the schema, no?

Reply:
Oh it's part of the schema yes, but I was wondering whether you need to specify the `server_default` argument in the `alter_column` function as well. It sounds like not though.

Reply:
Oh sorry, somehow I missed...like 80% of your comment. Let me double check.

Reply:
Well, all the timestamps currently have a `DEFAULT now()`. So I'm guessing this was mostly just trying to set NOT NULL since the default was already set.